scio-bigtable/src/main/scala/com/spotify/scio/bigtable/TableAdmin.scala

```diff
@@ -142,6 +143,75 @@ object TableAdmin {
     }
   }
 
+  /**
+   * Set cell expiration.
+   * Adds or modifies a cell expiration rule for the provided tables and column families.
+   *
+   * @param tablesAndColumnFamilies A map of tables and column families.  Keys are table names.
+   *                                Values are a list of column family names.
+   * @param cellExpiration The duration before which garbage collection of a cell may occur.
+   *                       Note: minimum granularity is second.
+   */
+  def setCellExpiration(bigtableOptions: BigtableOptions,
+                        tablesAndColumnFamilies: Map[String, List[String]],
+                        cellExpiration: Duration): Unit = {
+    val protoDuration = ProtoDuration.newBuilder.setSeconds(cellExpiration.getStandardSeconds)
+    val gcRule = GcRule.newBuilder.setMaxAge(protoDuration).build
+    setGcRule(bigtableOptions, tablesAndColumnFamilies, gcRule)
+  }
+
+  /**
+   * Set GcRule.
+   * Adds or modifies a GcRule for the provided tables and column families.
+   *
+   * @param tablesAndColumnFamilies A map of tables and column families.  Keys are table names.
+   *                                Values are a list of column family names.
+   * @param gcRule The gcRule to set on the provided tables and families.
+   */
+  private def setGcRule(bigtableOptions: BigtableOptions,
+                        tablesAndColumnFamilies: Map[String, List[String]],
+                        gcRule: GcRule): Unit = {
+    val project = bigtableOptions.getProjectId
+    val instance = bigtableOptions.getInstanceId
+    val instancePath = s"projects/$project/instances/$instance"
+
+    adminClient(bigtableOptions) { client =>
+      tablesAndColumnFamilies.foreach { case (table, columnFamilies) =>
+        val tablePath = s"$instancePath/tables/$table"
+        val tableInfo = client.getTable(GetTableRequest.newBuilder.setName(tablePath).build)
+
+        val modifications: List[Modification] =
+          columnFamilies
+            .filter { cf =>
```

Comment:
nit 🎨 i guess you could use partition here and that would push the logging to a better place?

Reply:
I gave it a shot in https://github.com/spotify/scio/pull/2054/commits/a774601c9208afc6f6612ccfb4809293faaf4d3e. Is that what you had in mind?